include/neml2/models/Model.h

```diff
@@ -30,13 +30,14 @@
 #include "neml2/base/Registry.h"
 #include "neml2/base/NEML2Object.h"
 #include "neml2/base/Factory.h"
+#include "neml2/solvers/NonlinearSystem.h"
 
 namespace neml2
 {
 /**
 Class that maps some input -> output, which is also the broader definition of constitutive model.
 */
-class Model : public NEML2Object, public LabeledAxisInterface
+class Model : public NEML2Object, public LabeledAxisInterface, public NonlinearSystem
```

Comment:
I guess it's fine for now, but we may want a `NonlinearMaterialModel` (as opposed to a simple model like linear elasticity that doesn't need to also be a `NonlinearSystem`.

Reply:
Agreed